sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1044,6 +1044,34 @@ class Analyzer(
           failOnOuterReference(p)
           p
       }
+
+      // SPARK-17348
+      // Looking for a potential incorrect result case.
+      // When a correlated predicate is a non-equality predicate
+      // it must be placed at the immediate child operator.
+      // Otherwise, the pull up of the correlated predicate
+      // will generate a plan with a different semantics
+      // which could return incorrect result.
+      var continue : Boolean = true
```

Comment:
those are not aggregates right? I am saying that we should ban cases in which we pull out a predicate from an aggregate for example `t1.c2 >= t2.c2` in the following example:
```sql
select t1.c1
from   t1
where  t1.c1 in (select max(t2.c1)
                 from   t2
                 where  t1.c2 >= t2.c2)
```